src/datasets/load.py

```diff
@@ -422,7 +422,11 @@ def _create_importable_file(
 def infer_module_for_data_files(
     data_files_list: DataFilesList, use_auth_token: Optional[Union[bool, str]] = None
 ) -> Optional[str]:
-    extensions_counter = Counter(suffix[1:] for filepath in data_files_list for suffix in Path(filepath).suffixes)
+    extensions_counter = Counter(
+        suffix[1:]
+        for filepath in data_files_list[: config.DATA_FILES_MAX_NUMBER]
```

Comment:
I think we could have a greater maximum number than 10 here. Maybe 200 ? It won't hurt performance but it will work better for users that store data files as well as other files (documentation, generation code etc.)